### PR TITLE
Added Binary Search [Javascript]

### LINKED ISSUE #295: Binary Search [JavaScript]
#### This issue is about [Binary Search](https://en.wikipedia.org/wiki/Binary_search_algorith).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [ ] Assign this issue to me (I want to work on this).
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.

### PR DESCRIPTION
**Pull request by Monal5031**
Fixes #295

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: binary_search/binary_search.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	/*
+	Performs a binary search iteratively
+	:param arr: List of elements to search from
+	:param item: Element to search for
+	:return: returns index if element found else -1
+	*/
+	let begin = 0;
+	let end = arr.length - 1;
+	while (begin <= end) {
+		let mid = (begin + end) / 2;
```
Comment (aashutoshrathi) [inline binary_search/binary_search.js]: Should we floor it ?

File: binary_search/binary_search.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	/*
+	Performs a binary search recursively
+	:param arr: List of elements to search from
+	:param item: Element to search for
+	:param begin: Left limit of array
+	:param end: Right limit of array
+	:return: returns index if element found else -1
+	*/
+	if (begin <= end) {
+		let mid = (begin + end) / 2;
```
Comment (aashutoshrathi) [inline binary_search/binary_search.js]: Should we floor it?

File: binary_search/binary_search.js (hunk 1 above)
Comment (aviaryan) [inline binary_search/binary_search.js]: Nice catch.

File: binary_search/binary_search.js (hunk 2 above)
Comment (Monal5031) [inline binary_search/binary_search.js]: Done.

File: binary_search/binary_search.js (hunk 1 above)
Comment (Monal5031) [inline binary_search/binary_search.js]: Done.

**Pull request merged**